wordpress-mobile/WordPress-Android
# App config

This PR introduces an AppConfig class that uses Firebase remote config to handle remote feature flags and experiments. I was thinking about several ways to implement the feature flags/experiments and in the end I decided that the best way is to have injectable classes for all the flags/experiments. These classes could be directly injected into the place where you need them and called in this fashion: `featureConfig.isEnabled()`. Leet me know what you think about this approach. The remote values are now cached the first time they are retrieved (but they are updated more often on the background). This should help us update the value if the user hasn't used the feature yet. 

I've added a tracking to when the values are retrieved for the first time. These events could be used in funnels to see whether the use has the flag or is in an experiment. I've added 2 methods to the `AnalyticsTrackerWrapper` that you can use if you want to add the tracking information to any event you're tracking.

I've updated play services and the messaging library in order to make the firebase remote config work. I've checked the change log and the play services seem to only have 1 breaking change and that's introducing AndroidX and Jetpack. 
The messaging library introduces a new way to handle Instance ID which I've implemented so all should be good there. 

Test setup:
- Notice that there are two example files - `ExampleExperimentConfig` and `TenorFeature`
- Create an experiment/feature for testing with a `remote field` set
- Inject this experiment or feature anywhere in your codebase and log the current value
- Preferably use a screen that's not in the main activity
- Add the default value to `remote_config_defaults` file

Test 1:
- Setup your experiment or feature
- Open the app
- Open the tested screen
- Check your log 
- You see that the value is the one you used in `remote_config_defaults`

Test 2:
- Setup your experiment or feature
- Go to the Firebase console
- Create a flag with the your previous `remote field`
- set a value different from the `remote_config_defaults` value
- Open the app
- Open the tested screen
- Check your log 
- You see the new value

Test 3:
- Setup your experiment or feature
- Go to the Firebase console
- Create a flag with the your previous `remote field`
- set a **value A** different from the `remote_config_defaults` value
- Open the app
- Open the tested screen
- Check your log 
- You see the new **value A**
- Change the flag value in the firebase console to **value B**
- Move the app to the background/foreground
- Open the tested screen
- You see that the **value A** is still used
- Kill the app
- Open the tested screen
- You see that the **value B** is used

PR submission checklist:
